File: src/main/java/edu/kit/iti/formal/stvs/model/common/VariableCategory.java
package edu.kit.iti.formal.stvs.model.common;

/**
 * The category of a variable.
 *
 * @author Philipp
 */
public enum VariableCategory {
  INPUT(VariableRole.ASSUME), OUTPUT(VariableRole.ASSERT),
  INOUT(VariableRole.ASSERT), LOCAL(VariableRole.ASSUME);

  private final VariableRole defaultRole;

  VariableCategory(VariableRole defaultRole) {
    this.defaultRole = defaultRole;
  }

  public VariableRole getDefaultRole() {
    return defaultRole;
  }
}

File: stvs/src/main/kotlin/edu/kit/iti/formal/stvs/model/common/VariableCategory.kt
package edu.kit.iti.formal.stvs.model.common

/**
 * The category of a variable.
 *
 * @author Philipp
 */
enum class VariableCategory(val defaultRole: VariableRole) {
    INPUT(VariableRole.ASSUME), OUTPUT(VariableRole.ASSERT),
    INOUT(VariableRole.ASSERT), LOCAL(VariableRole.ASSUME)
}